xclim/indices/_agro.py

```diff
@@ -1418,3 +1419,52 @@ def effective_growing_degree_days(
     egdd = aggregate_between_dates(deg_days, start=start, end=end, freq=freq)
 
     return to_agg_units(egdd, tas, op="delta_prod")
+
+
+@declare_units(tasmin="[temperature]")
+def hardiness_zones(
+    tasmin: xarray.DataArray, window: int = 30, method: str = "usda", freq: str = "YS"
+):
+    """Hardiness zones.
+
+    Hardiness zones are a categorization of the annual extreme temperature minima, averaged over a certain period.
+    The USDA defines 14 zones, each divided into two sub-zones, using steps of 5°F, starting at -65°F.
+    The Australian National Botanic Gardens define 7 zones, using steps of 5°C, starting at -15°C.
+
+    Parameters
+    ----------
+    tasmin : xr.DataArray
+        Minimum temperature.
+    window : int
+        The length of the averaging window, in years.
+    method : {'usda', 'anbg'}
+        Whether to return the American (`usda`) or the Australian (`anbg`) classification zones.
+    freq : str
+        Resampling frequency.
+
+    Returns
+    -------
+    xr.DataArray, [dimensionless]
+        {method} hardiness zones.
+        US sub-zones are denoted by using a half step. For example, Zone 4b is given as 4.5.
+        Values are given at the end of the averaging window.
+
+    References
+    ----------
+    :cite:cts:`usda_2012`
+    :cite:cts:`dawson_plant_1991`
+    """
+    tnmin = tasmin.resample(time=freq).min().rolling(time=window).mean()
+
+    if method.lower() == "usda":
+        zones = get_zones(tnmin, "-60 degF", "70 degF", "5 degF")
```

Comment:
```suggestion
        zones = 0.5 + get_zones(tnmin, "-60 degF", "70 degF", "5 degF") / 2
```
Non ?
That's what the docstring says and what feels more natural to me.

Reply:
I think we opted not to do this, since we add a coordinate that explains the zones in the returned DataArray.